app/scripts/controllers/client/ViewClientController.js

```diff
@@ -846,31 +846,85 @@
+                        Upload.upload({
+                            url: $rootScope.hostUrl + API_VERSION + '/clients/' + routeParams.id + '/documents',
+                            data: {
+                                name: 'clientSignature',
+                                description: 'client signature'
+                            },
+                            file: file
+                        }).then(function (imageData) {
+                            // to fix IE not refreshing the model
+                            if (!scope.$$phase) {
+                                scope.$apply();
+                            }
+                            $uibModalInstance.dismiss('cancel');
+                            route.reload();
+                        });
+                    }
+                    
+                };
+                
             };
+
             var ViewLargerClientSignature = function($scope,$uibModalInstance){
                 var loadSignature = function(){
                     http({
                         method: 'GET',
                         url: $rootScope.hostUrl + API_VERSION + '/clients/' + routeParams.id + '/documents'
                     }).then(function (docsData) {
-                        var docId = -1;
+                        $scope.docId = -1;
                         for (var i = 0; i < docsData.data.length; ++i) {
                             if (docsData.data[i].name == 'clientSignature') {
-                                docId = docsData.data[i].id;
-                                scope.signature_url = $rootScope.hostUrl + API_VERSION + '/clients/' + routeParams.id + '/documents/' + docId + '/attachment?tenantIdentifier=' + $rootScope.tenantIdentifier;
+                                $scope.docId = docsData.data[i].id;
+                                scope.signature_url = $rootScope.hostUrl + API_VERSION + '/clients/' + routeParams.id + '/documents/' + $scope.docId + '/attachment?tenantIdentifier=' + $rootScope.tenantIdentifier;
                             }
                         }
                         if (scope.signature_url != null) {
                             http({
                                 method: 'GET',
-                                url: $rootScope.hostUrl + API_VERSION + '/clients/' + routeParams.id + '/documents/' + docId + '/attachment?tenantIdentifier=' + $rootScope.tenantIdentifier
+                                url: $rootScope.hostUrl + API_VERSION + '/clients/' + routeParams.id + '/documents/' + $scope.docId + '/attachment?tenantIdentifier=' + $rootScope.tenantIdentifier
                             }).then(function (docsData) {
                                 $scope.largeImage = scope.signature_url;
                             });
                         }
                     });
                 };
                 loadSignature();
+                $scope.deleteSig = function () {
+                    http({
```

Comment:
There is no need to duplicate the code here, You had already defined a deleteSignature feature in your previous PR, why not just use that feature? what I mean is Instead of adding those functionalities in the View Signature modal, we could just make the view client modal open up the delete modal. eg: Here if you make changes like
```
$scope.deleteSig = function(){
       $uibModalInstance.dismiss('cancel') // Dismiss the current view signature modal
        scope.deleteSig() // Open the deleteSignature modal  which is already defined
}
```

This should open up the delete signature modal where the user can confirm to delete.

This will reduce a lot of duplicate and extra codes :)